scheduler/src/cook/rest/api.clj

```diff
@@ -954,12 +987,32 @@
                (not (contains? (get-gpu-models-on-pool (config/valid-gpu-models) pool-name) requested-gpu-model)))
       (throw (ex-info (str "The following GPU model is not supported: " requested-gpu-model) {})))))
 
+(defn validate-job-disk
+  "Validates that a job requesting disk is satisfying the following conditions:
+    - User can request size but not type; User can request both size and type; User cannot request only type and not size
+    - Requested type must be a valid type in config
+    - Requested size must be less than the max size in config"
+  [pool-name {:keys [disk]}]
+  (let [{disk-request :request disk-limit :limit requested-disk-type :type} disk
+        max-size (or (get-max-disk-size-on-pool (config/disk) pool-name) 10000000)
+        disk-types-on-pool (get-disk-types-on-pool (config/disk) pool-name)]
+    (when disk-limit
+      (do (when (> disk-request disk-limit)
+            (throw (ex-info (str "Disk request specified is greater than disk limit specified") disk)))
+          (when (> disk-limit max-size)
+            (throw (ex-info (str "Disk limit specified is greater than max disk size on pool") disk)))))
```

Comment:
Maybe cleaner and more obvious with:
  (when (< disk-request disk-limit max-size)  
       (.... "Disk resource settting error, We must have disk-request < disk-limit < max-size, but have "  {:disk-request disk-request :max-size ....})

Reply:
thanks for suggestion. done